cadquery/cq.py

```diff
@@ -3919,28 +3919,43 @@ def text(
         """
         Create a 3D text
 
-        :param str txt: text to be rendered
-        :param distance: the distance to extrude, normal to the workplane plane
+        :param txt: text to be rendered
+        :param fontsize: size of the font in model units
+        :param distance: the distance to extrude or cut, normal to the workplane plane
         :type distance: float, negative means opposite the normal direction
-        :param float fontsize: size of the font
-        :param boolean cut: True to cut the resulting solid from the parent solids if found.
-        :param boolean combine: True to combine the resulting solid with parent solids if found.
-        :param boolean clean: call :py:meth:`clean` afterwards to have a clean shape
-        :param str font: fontname (default: Arial)
-        :param str kind: font type (default: Normal)
-        :param str halign: horizontal alignment (default: center)
-        :param str valign: vertical alignment (default: center)
-        :return: a CQ object with the resulting solid selected.
-
-        extrude always *adds* material to a part.
-
-        The returned object is always a CQ object, and depends on whether combine is True, and
+        :param cut: True to cut the resulting solid from the parent solids if found
+        :param combine: True to combine the resulting solid with parent solids if found
+        :param clean: call :py:meth:`clean` afterwards to have a clean shape
+        :param font: font name
```

Comment:
What is the rationale for removing default values?

Reply:
They are shown in the signature on HTML docs. 
![screenshot2021-07-09-153054](https://user-images.githubusercontent.com/50230945/125030705-b0410980-e0ca-11eb-8073-98f8b7dbb9b9.png)
I'm not aware of anywhere that you could see this doc string and not the signature, but let me know if there is an IDE or something where it's a problem.

Reply:
OK, I'm not sure that it is better for the reader. I agree though that it'd be nice if they were automatically generated from the method definition.

Reply:
> it'd be nice if they were automatically generated from the method definition

Here is a way to do that (at least when using Sphinx's inbuilt autodoc-typehints):

```python
def process_docstring_add_defaults(app, what, name, obj, options, lines):
    import inspect
    import re
    if what == "method":
        sig = inspect.signature(obj)
        for idx in range(len(lines)):
            pattern = re.compile(r"^\s*:param\s(\w*):")
            match = pattern.search(lines[idx])
            if match is None:
                continue
            arg = match.group(1)
            if sig.parameters[arg].default != inspect.Parameter.empty:
                lines[idx] += f" (default is {sig.parameters[arg].default})"

def setup(app):
    app.connect("autodoc-process-docstring", process_docstring_add_defaults)
```

But I'm getting off topic for this PR.

Reply:
OK, maybe good to open it.